httpcache.go

```diff
@@ -350,9 +334,36 @@ type ctxKey string
 
 const getterContextCtxKey ctxKey = "getter_context"
 
+func parseCaddyfileGlobalOption(d *caddyfile.Dispenser) (interface{}, error) {
+	cfg = &Config{}
+	for d.Next() {
+		for d.NextBlock(0) {
+			switch d.Val() {
+			case "olric_config":
+				if !d.NextArg() {
+					return nil, d.ArgErr()
+				}
+
+				cfg.OlricConfig = d.Val()
+			}
+		}
+	}
+
+	return nil, nil
```

Comment:
This doesn't return the config; come to think of it I'm not sure if the extensible global options are wired up yet: https://github.com/caddyserver/caddy/issues/3634

Reply:
Yes, what I've done here is a hack, it's kind of abusing the hook, but it works pretty well for this very specific use case - creating a global immutable instance (I tested it extensively).
https://github.com/caddyserver/caddy/issues/3634 will be needed for most modules anyway, and will maybe allow to make something cleaner here, but it's not a blocker for this PR.

Reply:
Ah... that's very interesting. Willing to roll with it for now!